sdk/go/pulumi/run.go

```diff
@@ -81,7 +83,13 @@ func RunErr(body RunFunc, opts ...RunOption) error {
 	}
 	defer contract.IgnoreClose(ctx)
 
-	return RunWithContext(ctx, body)
+	err = RunWithContext(ctx, body)
+	// Log the error message
+	if err != nil {
+		err := ctx.Log.Error(fmt.Sprintf("an unhandled error occurred: program failed: \n%v", err), nil)
```

Comment:
Do we know who calls RunErr? This is a change in this function that now makes it log to ctx.Log; small change but still.

Reply:
Sounds like folks use it in mock tests e.g. https://github.com/pulumi/examples/blob/ca40203279f393c0c159dadcadc97c6007122997/testing-unit-go/main_test.go#L36

Reply:
We ask users to use RunErr e.g. in here https://www.pulumi.com/docs/guides/testing/unit/

Reply:
So the change might be ok if it surfaces ok in those mock testing examples, but I'd be a bit worried without double-checking; it might be just easier to arrange code so that RunErr does not have the changed behavior.